### PR TITLE
Do not rely on idm auth anymore. Add support for macaroons instead

### PR DESCRIPTION
**Pull request by frankban**

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/theblues-lib/207/

File: theblues/identity_manager.py [hunk 1]
```diff
@@ -20,28 +20,25 @@
...
-        self.auth = (idm_user, idm_password)
         self.timeout = timeout
 
-    def get_user(self, username):
+    def get_user(self, username, macaroons):
         """Fetch user data.
 
         Raise a ServerError if an error occurs in the request process.
 
         @param username the user's name.
```
Comment (jcsackett) [inline theblues/identity_manager.py:38]: Can you add the macaroon param here, given how often macaroons trip people up. :smile:

**Review by jcsackett — APPROVED** [review]

File: theblues/identity_manager.py (hunk 1 above)
Comment (frankban) [reply theblues/identity_manager.py:38]: Done.

**Comment by frankban** [comment]
Thanks for the review!
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/theblues-lib-merge

**Pull request merged**